src/server/identity/server/dex_web.go

```diff
@@ -58,11 +62,33 @@ func (w *dexWeb) stopWebServer() {
 	w.server = nil
 }
 
-// startWebServer must called while holding the write mutex
-func (w *dexWeb) startWebServer() {
+// startWebServer returns a new dex web server, and a boolean for whether it should be cached.
+func (w *dexWeb) startWebServer() (*dex_server.Server, bool) {
+	cache := true
 	storage, err := w.storageProvider.GetStorage(w.logger)
 	if err != nil {
-		return
+		return nil, false
```

Comment:
nit: would it make sense to log this error or something? Is it every meaningful?

Reply:
The storage provider logs the error right now. It would probably make sense to consolidate all the logging in one place - I think I can actually eliminate the storage provider altogether now that postgres is part of every deployment, so we don't need to start lazily.